samples: openthread: add esp platforms to the coap

wmrsouza opened the pull request:
Add the following esp platforms to the openthread coap sample:
    - esp32c5_devkitc/esp32c5/hpcore
    - esp32c6_devkitc/esp32c6/hpcore
    - esp32h2_devkitm
Add also the following esp platforms using rcp trhough uart:
    - esp32_devkitc/esp32/procpu
    - esp32s3_devkitc/esp32s3/procpu

wmrsouza committed:
samples: openthread: add esp platforms to the coap sample

Add the following platforms to the openthread coap sample:
    - esp32c5_devkitc/esp32c5/hpcore
    - esp32c6_devkitc/esp32c6/hpcore
    - esp32h2_devkitm
and adds the following platforms using rcp through hdlc over uart:
    - esp32_devkitc/esp32s3/procpu
    - esp32s3_devkitc/esp32s3/procpu

Signed-off-by: Marcio Ribeiro <marcio.ribeiro@espressif.com>

rlubos commented on samples/net/openthread/coap/overlay-ot-rcp-esp32.conf:
```diff
@@ -0,0 +1,35 @@
+#
```
Why does it need to be an overlay, can't it be a board-specific config in boards extension? It'd be applied automatically for the platform.

wmrsouza committed:
modules: openthread: add external memory support for heap and message pool

Add Kconfig options to allocate OpenThread dynamic memory and message
pool buffer from external memory on ESP platforms:
- OPENTHREAD_EXTERNAL_HEAP_SHARED_MULTI_HEAPOPENTHREAD_HEAP_EXTERNAL:
  routes otPlatCAlloc/otPlatFree through shared multi-heap to allocate
  from external memory.
- OPENTHREAD_MESSAGE_POOL_CUSTOM_SECTION: places the message pool
  buffer in a custom .ot_msg_pool linker section.
- Add .ot_msg_pool section to external RAM in the linker
  script for  ESP32, ESP32-C5, ESP32-S3, ESP32-S3

This allows offloading large OpenThread buffers to PSRAM, freeing
internal SRAM for other uses.

Signed-off-by: Marcio Ribeiro <marcio.ribeiro@espressif.com>

wmrsouza commented on samples/net/openthread/coap/sample.yaml:
```diff
@@ -17,3 +17,32 @@ tests:
+      - esp32c6_devkitc/esp32c6/hpcore
+      - esp32h2_devkitm
+    extra_args:
+      - CONFIG_OT_COAP_SAMPLE_SERVER=y
```
I only see benefits doing this.
